packages/eslint-plugin/src/rules/no-extra-semi.ts

```diff
@@ -0,0 +1,31 @@
+import baseRule from 'eslint/lib/rules/no-extra-semi';
+import * as util from '../util';
+
+type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
+type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
+
+export default util.createRule<Options, MessageIds>({
+  name: 'no-extra-semi',
+  meta: {
+    type: 'suggestion',
+    docs: {
+      description: 'Disallow unnecessary semicolons',
+      category: 'Possible Errors',
+      recommended: 'error',
```

Comment:
nit - must be false for now. we can make it recommended in the next breaking release.